File: javascript algorithms/검색,정렬/quickSort.py
import random


def qsort(data):
    if len(data) < 1:
        return data

    left, right = list(), list()
    pivot = data[0]

    for index in range(len(data)):
        if data[index] < pivot:
            left.append(data[index])
        elif data[index] > pivot:
            right.append(data[index])

    return qsort(left) + [pivot] + qsort(right)


data_list = random.sample(range(100), 10)
qsort(data_list)

File: algorithm/검색,정렬/quickSort.js
function qsort(data) {
  if (data.length < 1) {
    return data;
  }

  let left = [];
  let right = [];
  let pivot = data[0];

  for (let index in data) {
    if (data[index] < pivot) {
      left.push(data[index]);
    } else if (data[index] > pivot) {
      right.push(data[index]);
    }
  }
  return qsort(left).concat([pivot]).concat(qsort(right));
}
console.log(qsort([12, 4, 35, 54, 65, 73, 23, 4, 54]));
// qsort([12, 4, 35, 54, 65, 73, 23, 4, 54]);
